src/main/java/com/learning/springboot/service/TransactionService.java

```diff
@@ -50,21 +53,40 @@ public TransactionAE getTransactionById(int id) {
 		TransactionAE transactionAE = convert(transactionDE);
 		return transactionAE;
 	}
-	
+
 	public List<TransactionAE> getActiveTransaction() {
-		
+
 		List<TransactionDE> deList = (List<TransactionDE>) transactionRepository.findAllByDeletedFalse();
 		List<TransactionAE> aeList = new ArrayList<TransactionAE>();
-		for(int i=0; i< deList.size();i++) {
+		for (int i = 0; i < deList.size(); i++) {
 			TransactionDE transactionDE = deList.get(i);
 			TransactionAE transactionAE = convert(transactionDE);
 			aeList.add(transactionAE);
 		}
 		return aeList;
 	}
-	
+
+	public List<TransactionAE> getAllTransactionsByGivenCategory(CategoryAE ae) {
+		List<TransactionDE> deList = (List<TransactionDE>) transactionRepository.findAllByCategoryId(ae.getId());
+		List<TransactionAE> aeList = new ArrayList<TransactionAE>();
+		Calendar cal1 = Calendar.getInstance();
+		Calendar cal2 = Calendar.getInstance();
+		for (int i = 0; i < deList.size(); i++) {
+			TransactionDE transactionDE = deList.get(i);
+			cal1.setTime(transactionDE.getDate());
+			cal2.setTime(new Date());
+			if (cal1.get(Calendar.YEAR) == cal2.get(Calendar.YEAR)) {
```

Comment:
This algorithm is very inefficient. It will get all the transactions for any category, which could be 10000+. Instead, ask database to return only those transactions which lies in current month. Use `between` operator in your repository method. E.g. https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.query-creation, https://www.baeldung.com/spring-data-jpa-query-by-date